### PR TITLE
chore: upgrade core crypto 1.0.0.rc12

### PR DESCRIPTION
**Pull request by typfel**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Upgrade core crypto to `1.0.0-rc.2`

This version comes with significant API breakage where most methods have become suspendible.

- [x] Update APIs after turning suspendible
- [x] Refactor `ProteusClientProvider` and `MLSClientProvider` due to changes in how CoreCrypto is instantiated.
- [x] Handle new errors `SelfCommitIgnored`, `DuplicateMessage `, `UnmergedPendingGroup`
- [x] Process buffered messages when merging group after joining via ext commit (requires refactoring will be done separately)

#### Test Coverage

- [x] I have added automated test to this contribution

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by typfel** [comment]
Tests sometimes never finish which might be caused by this https://github.com/mozilla/uniffi-rs/issues/1669

File: util/src/androidInstrumentedTest/kotlin/com/wire/kalium/util/FileTestHelper.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.createTempDirectory
+
+actual object FileTestHelper {
+    actual fun createRandomDirectory(): String {
+        return createTempDirectory("testDirectory").absolutePathString()
+    }
+
+    actual fun createRandomFileAt(path: String) {
+        kotlin.io.path.createTempFile(directory = Path(path), prefix = "testFile")
```
Comment (borichellow) [inline util/src/androidInstrumentedTest/kotlin/com/wire/kalium/util/FileTestHelper.kt:31]: why not import this fun, so you'll have only `createTempFile(directory = Path(path), prefix = "testFile")` here?

File: util/src/jvmTest/kotlin/com/wire/kalium/util/FileTestHelper.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import kotlin.io.path.absolutePathString
+import kotlin.io.path.createTempDirectory
+
+actual object FileTestHelper {
+    actual fun createRandomDirectory(): String {
+        return createTempDirectory("testDirectory").absolutePathString()
+    }
+
+    actual fun createRandomFileAt(path: String) {
+        kotlin.io.path.createTempFile(directory = Path(path), prefix = "testFile")
```
Comment (borichellow) [inline util/src/jvmTest/kotlin/com/wire/kalium/util/FileTestHelper.kt:31]: same as above

**Review by borichellow — APPROVED** [review]
Nice one! 
Just tiny suggestion, that is not much important

**Review by mchenani — APPROVED** [review]
🔥

File: cryptography/build.gradle.kts [hunk 3]
```diff
@@ -101,7 +101,10 @@ kotlin {
             dependencies {
                 implementation(libs.cryptoboxAndroid)
                 implementation(libs.androidCrypto)
-                implementation(libs.coreCryptoAndroid)
+                implementation(libs.coreCryptoAndroid.get().let { "${it.module}:${it.versionConstraint.requiredVersion}" }) {
```
Comment (yamilmedina) [inline cryptography/build.gradle.kts:104]: Cool trick! 🚀

**Review by yamilmedina — APPROVED** [review]

**Pull request merged**